### PR TITLE
Add flake8 check for pep8 naming

### PR DESCRIPTION
**Pull request by jackwilsdon**
Add flake8 check for pep8 naming using [pep8-naming](https://github.com/PyCQA/pep8-naming).

In it's current state (as of 698203eb847020194987eef63ad7da052eb86ab0), this check is not passed by beets.

Here are the offending files and line numbers:

```
beets/logging.py:129:5: N802 function name should be lowercase
beets/dbcore/db.py:212:9: N804 first argument of a classmethod should be named 'cls'
beets/dbcore/db.py:277:9: N804 first argument of a classmethod should be named 'cls'
beets/dbcore/query.py:149:9: N804 first argument of a classmethod should be named 'cls'
beets/dbcore/query.py:844:9: N805 first argument of a method should be named 'self'
beets/autotag/hooks.py:300:9: N805 first argument of a method should be named 'self'
beets/util/artresizer.py:152:9: N805 first argument of a method should be named 'self'
beets/util/artresizer.py:157:9: N805 first argument of a method should be named 'self'
beets/util/artresizer.py:232:5: N802 function name should be lowercase
beets/util/artresizer.py:251:5: N802 function name should be lowercase
beets/util/confit.py:139:9: N804 first argument of a classmethod should be named 'cls'
beetsplug/bucket.py:158:5: N806 variable in function should be lowercase
beetsplug/bucket.py:165:17: N806 variable in function should be lowercase
beetsplug/bucket.py:166:17: N806 variable in function should be lowercase
beetsplug/export.py:122:9: N804 first argument of a classmethod should be named 'cls'
beetsplug/fuzzy.py:29:9: N804 first argument of a classmethod should be named 'cls'
beetsplug/fuzzy.py:33:9: N806 variable in function should be lowercase
test/_common.py:170:9: N802 function name should be lowercase
test/_common.py:174:9: N802 function name should be lowercase
test/test_art.py:521:9: N802 function name should be lowercase
test/test_art.py:534:9: N802 function name should be lowercase
test/test_autotag.py:943:9: N806 variable in function should be lowercase
test/test_convert.py:43:9: N802 function name should be lowercase
test/test_convert.py:53:9: N802 function name should be lowercase
test/test_datequery.py:65:9: N802 function name should be lowercase
test/test_datequery.py:72:9: N802 function name should be lowercase
test/test_edit.py:74:9: N802 function name should be lowercase
test/test_edit.py:136:9: N802 function name should be lowercase
test/test_importadded.py:70:9: N802 function name should be lowercase
test/test_importadded.py:78:9: N802 function name should be lowercase
test/test_importadded.py:82:9: N802 function name should be lowercase
test/test_importer.py:128:9: N806 variable in function should be lowercase
test/test_lastgenre.py:170:9: N806 variable in function should be lowercase
test/test_lyrics.py:368:9: N806 variable in function should be lowercase
test/test_lyrics.py:374:9: N806 variable in function should be lowercase
test/test_mediafile.py:171:9: N802 function name should be lowercase
test/test_mediafile.py:180:9: N802 function name should be lowercase
test/test_mediafile.py:663:9: N802 function name should be lowercase
test/test_mpdstats.py:46:9: N806 variable in function should be lowercase
test/test_mpdstats.py:66:9: N802 function name should be lowercase
test/test_permissions.py:41:9: N803 argument name should be lowercase
test/test_permissions.py:61:9: N802 function name should be lowercase
test/test_permissions.py:61:9: N803 argument name should be lowercase
test/test_query.py:38:9: N802 function name should be lowercase
test/test_query.py:42:9: N802 function name should be lowercase
test/test_query.py:808:9: N802 function name should be lowercase
test/test_smartplaylist.py:91:9: N806 variable in function should be lowercase
test/test_ui.py:236:9: N806 variable in function should be lowercase
test/test_ui.py:237:9: N806 variable in function should be lowercase
test/test_ui.py:244:9: N806 variable in function should be lowercase
test/test_ui.py:245:9: N806 variable in function should be lowercase
```

Looking through the code (mostly tests), there does seem to be a mix of camel case and snake case, as well as some inconsistencies with first argument naming.

I can imagine fixing these causing a lot of merge conflicts; is it worth it to follow pep8?

### CODE REVIEW

**Comment by jackwilsdon** [comment]
Oh dear, look who's back @sampsyo!

**Comment by sampsyo** [comment]
Noooooo! :angry: I filed an issue here; we'll see what they say: https://github.com/codecov/support/issues/162

At the moment https://codecov.io is down, so it's tricky to investigate.

**Comment by sampsyo** [comment]
Good idea—it would be great to enforce this automatically. It's something I often find myself asking people to change manually, so we should work toward merging this, even if it will take some effort.

There are also cases in this list where the naming violations are intentional. For example:

* [logging.py:129](https://github.com/beetbox/beets/blob/09bc2504133150ab19eda133565b9e513ac81544/beets/logging.py#L129): This matches the stdlib `logging` module, which predates PEP8.
* [Test assertion methods](https://github.com/beetbox/beets/blob/master/test/_common.py#L170) also match the standard library, sadly.
* [This shortcut](https://github.com/beetbox/beets/blob/09bc2504133150ab19eda133565b9e513ac81544/test/test_smartplaylist.py#L91) is a fairly benign renaming, and it refers to a class, so a capital letter makes sense to me.

So, in short, we'll need to use our judgment. Some of those `self` and `cls` naming problems are quite egregious, though!

**Comment by sampsyo** [comment]
On the Codecov issue: it looks like the file may need to be named `yml`, not `yaml`, which is insane? I renamed it, so maybe this will fix our problems.

Obviously my fault, but if I ran Codecov, I wouldn't display this message if `codecov.yml` is missing altogether:

> Your codecov.yml build successfully

**Comment by jackwilsdon** [comment]
So do you think I should start fixing these failures and adding `# noqa` to any lines that are valid?

**Comment by sampsyo** [comment]
Yes! Fixing the name problems in this PR seems right to me.

**Comment by jackwilsdon** [comment]
What should I do with issues in confit? Should I make the changes here or in the main repo?

**Comment by jackwilsdon** [comment]
I left the check failure in confit, as I'm not sure whether I should change it here or in [it's repository](https://github.com/sampsyo/confit).

Another question; is there a reason we aren't just pulling in confit from pip? Have we made some changes in beets that are not suitable for confit upstream?

**Comment by sampsyo** [comment]
You make an astute observation about Confit. The only reasons we haven't taken it as a dependency are the logistics: there's some friction for OS packagers to catalog a new dependency, and we'd need to stabilize the Confit API. The latter entails starting a Changelog, and because the Confit name is taken on PyPI, changing the module name to `confuse`.

It's definitely a worthwhile endeavor! It will just take a little more effort than deleting the file. Perhaps we should do the same thing for Confit, MediaFile, and dbcore simultaneously?

In the mean time, we can just fix the problem in both repositories simultaneously. This is not awesome, but it has worked for us so far.

**Comment by jackwilsdon** [comment]
Alright, I'll fix it for them now 👍. I'll create a new PR for modularising beets, I know it will take a while but it will be worth having it on record 😄.

File: beets/autotag/hooks.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def __init__(self):
         self._penalties = {}
 
     @LazyClassProperty
-    def _weights(cls):
+    def _weights(self):
```
Comment (sampsyo) [inline beets/autotag/hooks.py]: Hmm; this one is actually a class method—the style checker just doesn't know that!

File: beets/dbcore/query.py [hunk 2]
```diff
@@ -841,8 +841,8 @@ def is_slow(self):
 
 class NullSort(Sort):
     """No sorting. Leave results unsorted."""
-    def sort(items):
-        return items
+    def sort(self):
+        return self
```
Comment (sampsyo) [inline beets/dbcore/query.py]: This one should be:

```py
def sort(self, items):
    return items
```

That is, it's just missing an unused `self` parameter; it's not misnamed.

File: beets/autotag/hooks.py (hunk 1 above)
Comment (jackwilsdon) [inline beets/autotag/hooks.py]: Aha, I didn't notice that! I'll revert it and add a `# noqa`.

File: beets/dbcore/query.py (hunk 2 above)
Comment (jackwilsdon) [inline beets/dbcore/query.py]: Ah, makes sense. I'll fix it now.

File: test/test_autotag.py [hunk 3]
```diff
@@ -940,13 +940,13 @@ class EnumTest(_common.TestCase):
...
-        self.assertGreater(OrderedEnumTest.b, OrderedEnumTest.a)
-        self.assertGreater(OrderedEnumTest.c, OrderedEnumTest.a)
-        self.assertGreater(OrderedEnumTest.c, OrderedEnumTest.b)
+        ordered_enum = match.OrderedEnum('OrderedEnumTest', ['a', 'b', 'c'])
+        self.assertLess(ordered_enum.a, ordered_enum.b)
+        self.assertLess(ordered_enum.a, ordered_enum.c)
+        self.assertLess(ordered_enum.b, ordered_enum.c)
+        self.assertGreater(ordered_enum.b, ordered_enum.a)
+        self.assertGreater(ordered_enum.c, ordered_enum.a)
+        self.assertGreater(ordered_enum.c, ordered_enum.b)
```
Comment (sampsyo) [inline test/test_autotag.py]: I'm not sure about this one—`OrderedEnumTest` is actually a class, not an instance. This one doesn't matter too much either way though.

**Comment by sampsyo** [comment]
Woohoo; thanks! I think this is ready to merge. :rocket: :sparkles:

File: test/test_autotag.py (hunk 3 above)
Comment (jackwilsdon) [inline test/test_autotag.py]: Yeah, this is also kind of present in [`test/test_smartplaylist.py`]:

```python
sort = FixedFieldSort  # short cut since we're only dealing with this
asseq(sorts["no_sort"], NullSort())
asseq(sorts["one_sort"], sort(u'year'))
asseq(sorts["only_empty_sorts"], None)
asseq(sorts["one_non_empty_sort"], sort(u'year'))
asseq(sorts["multiple_sorts"],
    MultipleSort([sort('year'), sort(u'genre', False)]))
asseq(sorts["mixed"],
    MultipleSort([sort('year'), sort(u'genre'), sort(u'id', False)]))
```

It was originally named `S` instead of `sort`.

Comment (sampsyo) [inline test/test_autotag.py]: Sure! I don't have a strong feeling either way, so I trust your judgment. :+1:

**Comment by jackwilsdon** [comment]
Well it seems we just opened a whole can of worms! Merge conflicts, merge conflicts everywhere... 😆
